### ISSUE TITLE
ShortVariable rule features

### ISSUE THREAD

**Issue by dylan-tao** [pmd/pmd#4415]
**Affects PMD Version:**  
6.53.0

**Rule:** 

Please provide the rule name and a link to the rule documentation:
<https://pmd.sourceforge.io/pmd-6.53.0/pmd_rules_java_codestyle.html#shortvariable#ShortVariable>

**Description:**
equals(Object o) and T t intercepted.

**Code Sample demonstrating the issue:**

```java
@Override
  1. public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }

    Dict dict = (Dict) o;

    return code == dict.code;
  }

2. boolean insert(T t);

```

**Expected outcome:**

 idea auto generated equals(Object o) and T t, please ignore this，thanks.

**Running PMD through:** *[Maven Plugin]*

**Comment by dylan-tao** [comment]
I have a good idea, ShortVariable filter by Field、Formal、Local、For can custom setting, extend LongVariable in the same way.

<rule ref="category/java/codestyle.xml/ShortVariable">
    <properties>
        <property name="minimum" value="3" />
        <property name="Field" value="true" />
        <property name="Formal" value="false" />
        <property name="Local" value="true" />
        <property name="For" value="false" />
    </properties>
</rule>

PS：
public class Something {
    int reallyLongIntName = -3;             // VIOLATION - Field
    public static void main( String argumentsList[] ) { // VIOLATION - Formal
        int otherReallyLongName = -5;       // VIOLATION - Local
        for (int interestingIntIndex = 0;   // VIOLATION - For
             interestingIntIndex < 10;
             interestingIntIndex ++ ) {
    }
}

**Comment by dylan-tao** [comment]
> I have a good idea, ShortVariable filter by Field、Formal、Local、For can custom setting, extend LongVariable in the same way.
> 
> ```
> <rule ref="category/java/codestyle.xml/ShortVariable">
>     <properties>
>         <property name="minimum" value="3" />
>         <property name="Field" value="true" />
>         <property name="Formal" value="false" />
>         <property name="Local" value="true" />
>         <property name="For" value="false" />
>     </properties>
> </rule>
> ```
> 
> PS：
> 
> ```
> public class Something {
>     int reallyLongIntName = -3;             // VIOLATION - Field
>     public static void main( String argumentsList[] ) { // VIOLATION - Formal
>         int otherReallyLongName = -5;       // VIOLATION - Local
>         for (int interestingIntIndex = 0;   // VIOLATION - For
>              interestingIntIndex < 10;
>              interestingIntIndex ++ ) {
>     }
> }
> ```

@adangel How's the progress going？

**Comment by adangel** [comment]
Thanks for the suggestion.
However, I think, filter by Formal parameter is not good - it would essentially disable the rule for all formal parameters, whether it's your own method or not (e.g. inherited).
Also note, that the rule already ignores variable names in for loops.

I'd prefer to have instead a parameter "ignoreInheritedFormalParameters" (or some better name :)) - which ignores short formal parameter names, if the method is overridden. Note, that you can still change the parameter names, you are not forced to use the same names as in the parent class. That means, you can still fix such issues without any problem.

Also note, that IDEs tend to generate such one-letter parameter names if they don't have access to the source code of the parent class - which is a general project configuration error. Assuming the parent class uses good names for the formal parameters, the IDE will generate the overridden method with the same good parameter names.

For that reasons, I would keep this new parameter "false" by default, so that _all_ formal parameters are considered by default. Otherwise you would never tend to the root cause (wrong project setting, bad names in the parent class).
Also note, that IMHO a good habit would be to immediately give the parameters a better name after the IDE has inserted the boilerplate. You anyway need to understand what these parameters do in order to write a implementation.

**Comment by dylan-tao** [comment]
> Thanks for the suggestion. However, I think, filter by Formal parameter is not good - it would essentially disable the rule for all formal parameters, whether it's your own method or not (e.g. inherited). Also note, that the rule already ignores variable names in for loops.
> 
> I'd prefer to have instead a parameter "ignoreInheritedFormalParameters" (or some better name :)) - which ignores short formal parameter names, if the method is overridden. Note, that you can still change the parameter names, you are not forced to use the same names as in the parent class. That means, you can still fix such issues without any problem.
> 
> Also note, that IDEs tend to generate such one-letter parameter names if they don't have access to the source code of the parent class - which is a general project configuration error. Assuming the parent class uses good names for the formal parameters, the IDE will generate the overridden method with the same good parameter names.
> 
> For that reasons, I would keep this new parameter "false" by default, so that _all_ formal parameters are considered by default. Otherwise you would never tend to the root cause (wrong project setting, bad names in the parent class). Also note, that IMHO a good habit would be to immediately give the parameters a better name after the IDE has inserted the boilerplate. You anyway need to understand what these parameters do in order to write a implementation.

My brother, your way seems better, like.  can it be implemented in 7.0.0?
